src/Microsoft.TestPlatform.CoreUtilities/Utilities/JobQueue.cs

```diff
@@ -197,7 +197,14 @@ public void Flush()
     /// </summary>
     public void Dispose()
     {
-        if (_isDisposed)
+        Dispose(true);
+        GC.SuppressFinalize(this);
+    }
+
+    protected virtual void Dispose(bool disposing)
+    {
+        if (_isDisposed
```

Comment:
This is the only class that checks _isDisposed, why the other classes don't use/check that?

Reply:
I haven't changed the actual implementation of the `Dispose`, but only applied the pattern. Preventing multiple dispose is not always needed.

Reply:
>   Preventing multiple dispose is not always needed. 

Same as suppressing finalization 😁 Yet you still enforce it.

Reply:
I have addressed the comment as there were some merge conflicts.